src/MailAddress.php

```diff
@@ -7,21 +7,46 @@
 /**
  * @licence GNU GPL v2+
  * @author Christoph Fischer < christoph.fischer@wikimedia.de >
+ * @author Kai Nissen < kai.nissen@wikimedia.de >
  */
 class MailAddress {
-	public $userName;
-	public $domain;
+	private $userName;
+	private $domain;
+	private $displayName;
 
-	public function __construct( string $userName, string $domain ) {
-		$this->userName = $userName;
-		$this->domain = $domain;
+	public function __construct( string $emailAddress, string $displayName = '' ) {
```

Comment:
Initially I added a static function like that, but then I figured that it is actually quite unlikely to construct a MailAddress by passing a split email address. You're right about the $displayName, though. The only user would still be the 'operator', so maybe having a value object for it is unnecessary.